### ISSUE TITLE
Line Mode Browser OpenSSL

### ISSUE THREAD

**Issue by Lucas-mother3** [w3c/libwww#11]
Hello but yeah, like I said I couldn't access Google on Line Mode Browser. I have libwww with openssl installed. Any fix?

**Comment by jkbzh** [comment]
Sorry libwww is unmaintained since some years ago. No one has proposed to take over. I'd advise to try out curl or libhttp, according to your needs.

**Comment by Lucas-mother3** [comment]
> Sorry libwww is unmaintained since some years ago. No one has proposed to take over. I'd advise to try out curl or libhttp, according to your needs.

I understand that this software is unmaintained few years ago but i was curious about the second web browser that ever existed. I wanted to try it for myself and i want to view `https` websites. The Line Mode Browser 2013 project is down, so I went to this rabbit hole of unmaintained software and some patches with side of hibbie jibbie along the way. Can I at least give me an idea of how to get openssl working on Line Mode Browser and access Google?

**Comment by ericprud** [comment]
That could involve some engineering; could you set up an HTTP server to test drive LineMode?
That said, it's C and so is openssl so it's doable, with some effort.

**Comment by Lucas-mother3** [comment]
> That could involve some engineering; could you set up an HTTP server to test drive LineMode? That said, it's C and so is openssl so it's doable, with some effort.

What do you mean about hosting stuff tho? How should I do it in simpler terms?

**Comment by ericprud** [comment]
I'm not sure what your objective is, but if it's to try browsing a bit with LineMode, you could set up an HTTP server with some random content. It's not the same as using it LineMode anger in your daily life, but it would at least give a feel for the UX.

**Comment by Lucas-mother3** [comment]
> I'm not sure what your objective is, but if it's to try browsing a bit with LineMode, you could set up an HTTP server with some random content. It's not the same as using it LineMode anger in your daily life, but it would at least give a feel for the UX.

I tried browsing the W3M website, that was only the thing I browsed (i dunno about http without TLS/SSL in this day and age). And yeah, unlike Links or Lynx, I might have a learning curve but I understand the UX pretty well. My objective is to access Google, Wikipedia and my text version of my site (they all have TLS/SSL).

**Comment by ericprud** [comment]
I haven't touched libwww since Feb 97, but if I were as motivated as you seem to be, I'd set up a little HTTP server, run LineMode in a debugger, and set some break points around [HTBrowse](https://github.com/w3c/libwww/blob/master/LineMode/src/HTBrowse.c). That might be a good way to start untangling how to add SSL support so you can interact with an HTTPS server.

**Comment by Lucas-mother3** [comment]
> I haven't touched libwww since Feb 97, but if I were as motivated as you seem to be, I'd set up a little HTTP server, run LineMode in a debugger, and set some break points around [HTBrowse](https://github.com/w3c/libwww/blob/master/LineMode/src/HTBrowse.c). That might be a good way to start untangling how to add SSL support so you can interact with an HTTPS server.

I understand but can you please clarify with that one. Like the necessary commands and stuff

**Comment by Lucas-mother3** [comment]
Actually I built libwww with OpenSSL with patches to work on modern systems.

The patch for OpenSSL 1.1 compatibility was in http://cvsweb.netbsd.org/bsdweb.cgi/pkgsrc/www/libwww/patches/patch-Library_src_SSL_HTSSL.c?hideattic=0

The patch to detect OpenSSL: http://ix.io/3UP5

**Comment by Lucas-mother3** [comment]
So yeah after building libwww with OpenSSL, i tried Line Mode Browser and Google doesn't work. I dunno about how to debug it or something, heck I didn't know that much C language.

**Comment by Lucas-mother3** [comment]
> I haven't touched libwww since Feb 97, but if I were as motivated as you seem to be, I'd set up a little HTTP server, run LineMode in a debugger, and set some break points around [HTBrowse](https://github.com/w3c/libwww/blob/master/LineMode/src/HTBrowse.c). That might be a good way to start untangling how to add SSL support so you can interact with an HTTPS server.

Addendum: How do I debug it by myself

**Comment by Lucas-mother3** [comment]
@ericprud so i tried running `www -v http://google.com` and what i found was in this log:
[log.txt](https://github.com/w3c/libwww/files/8501080/log.txt)

**Comment by Lucas-mother3** [comment]
So I dug around the source and found HTBrowse.c, and since I dunno much about C except for patching suckless utilities, I couldn't do much, unless if there's a way to implement HTTPS on the main Line Mode Browser module.

**Comment by Lucas-mother3** [comment]
@ericprud @jkbzh so i tried tilde.club and it worked fine since it was HTTP.
